x/dex/module_test.go

```diff
@@ -170,6 +170,163 @@ func TestEndBlockMarketOrder(t *testing.T) {
+			OrderType:         types.OrderType_LIMIT,
+			PositionDirection: types.PositionDirection_SHORT,
+			Data:              "{\"position_effect\":\"Open\",\"leverage\":\"1\"}",
+		},
+	)
+	dexkeeper.MemState.GetDepositInfo(ctx, utils.ContractAddress(contractAddr.String())).Add(
+		&dexcache.DepositInfoEntry{
+			Creator: testAccount.String(),
+			Denom:   "uusdc",
+			Amount:  sdk.MustNewDecFromStr("2000000"),
+		},
+	)
+
+	ctx = ctx.WithBlockHeight(1)
+	testApp.EndBlocker(ctx, abci.RequestEndBlock{})
+	_, found := dexkeeper.GetLongBookByPrice(ctx, contractAddr.String(), sdk.MustNewDecFromStr("1"), pair.PriceDenom, pair.AssetDenom)
+	require.True(t, found)
+	_, found = dexkeeper.GetLongBookByPrice(ctx, contractAddr.String(), sdk.MustNewDecFromStr("2"), pair.PriceDenom, pair.AssetDenom)
+	require.True(t, found)
+	_, found = dexkeeper.GetShortBookByPrice(ctx, contractAddr.String(), sdk.MustNewDecFromStr("3"), pair.PriceDenom, pair.AssetDenom)
+	require.True(t, found)
+
+	dexkeeper.MemState.Clear()
+	dexkeeper.MemState.GetBlockOrders(ctx, utils.ContractAddress(contractAddr.String()), utils.GetPairString(&pair)).Add(
+		&types.Order{
+			Id:                4,
+			Account:           testAccount.String(),
+			ContractAddr:      contractAddr.String(),
+			Price:             sdk.MustNewDecFromStr("2"),
+			Quantity:          sdk.MustNewDecFromStr("2"),
+			PriceDenom:        pair.PriceDenom,
+			AssetDenom:        pair.AssetDenom,
+			OrderType:         types.OrderType_LIMIT,
+			PositionDirection: types.PositionDirection_SHORT,
+			Data:              "{\"position_effect\":\"Open\",\"leverage\":\"1\"}",
+		},
+	)
+	dexkeeper.MemState.GetBlockOrders(ctx, utils.ContractAddress(contractAddr.String()), utils.GetPairString(&pair)).Add(
+		&types.Order{
+			Id:                5,
+			Account:           testAccount.String(),
+			ContractAddr:      contractAddr.String(),
+			Price:             sdk.MustNewDecFromStr("3"),
+			Quantity:          sdk.MustNewDecFromStr("1"),
+			PriceDenom:        pair.PriceDenom,
+			AssetDenom:        pair.AssetDenom,
+			OrderType:         types.OrderType_LIMIT,
+			PositionDirection: types.PositionDirection_LONG,
+			Data:              "{\"position_effect\":\"Open\",\"leverage\":\"1\"}",
+		},
+	)
+
+	ctx = ctx.WithBlockHeight(2)
+	testApp.EndBlocker(ctx, abci.RequestEndBlock{})
+
+	_, found = dexkeeper.GetLongBookByPrice(ctx, contractAddr.String(), sdk.MustNewDecFromStr("2"), pair.PriceDenom, pair.AssetDenom)
+	require.False(t, found)
+	_, found = dexkeeper.GetLongBookByPrice(ctx, contractAddr.String(), sdk.MustNewDecFromStr("1"), pair.PriceDenom, pair.AssetDenom)
+	require.True(t, found)
+	_, found = dexkeeper.GetShortBookByPrice(ctx, contractAddr.String(), sdk.MustNewDecFromStr("3"), pair.PriceDenom, pair.AssetDenom)
+	require.True(t, found)
+
+	matchResults, _ := dexkeeper.GetMatchResultState(ctx, contractAddr.String(), 2)
+	require.Equal(t, 2, len(matchResults.Orders))
+	require.Equal(t, 4, len(matchResults.Settlements))
```

Comment:
can we check if correct amount of price=3 entries are remaining here

Reply:
sounds good, added one for the long book at price=3, there already is a check for the short book at price=3.